### ISSUE TITLE
Per-Subject Retention

### ISSUE THREAD

**Issue by cliffburdick** [nats-io/jetstream#317]
Hi, I have a use case where I have N subjects that I want the retention for each of those subjects to be the same, but not a global policy for all. The use case would be something like IoT where I have 30,000 different publishers, and I only want the last N messages from each publisher. Currently the stream would be the lowest granularity for retention, but that would only set a retention policy for all publishers in that stream.

**Comment by ripienaar** [comment]
I agree this would be needed, I have a similar problem in my projects around node metadata, node metrics, monitoring checks etc.

Added to our project as a backlog item.

**Comment by tpihl** [comment]
This would solve things for us also.

Maybe doesn't need to be individually set, maybe only either set a limitsPerSubjectPolicy where each limit is calculated per subject instead of the full stream. Maybe also to combine with a "total" policy just as today?
